frontend/week4/README.md

```diff
@@ -0,0 +1,131 @@
+# DevClub Summer of Code 2023
+
+## Frontend Development: Week 4   
+
+Welcome to the final stride of the DSoC '23 Frontend track! 
+
+In this task, you will be querying a backend service to get data and render it on the React frontend, thus completing the basic structure of any modern frontend application.
+
+This task will allow you to understand React hooks `useState()` and `useEffect()` in more depth, how state management works, and how to handle network requests and parse the output. 
+
+## Understanding the API   
+
+Applications communicate via APIs because they are flexible and make software development modular. Go to this [API specification page][0] and just play around with it. Read about [endpoints][2] and different [HTTP request types][3], what parameters they require. You should also read about [JSON format][1], the de-facto format for transferring data on the web.   
+
+The API contains CRUD (Create, Read, Update, Delete) endpoints for a Product model. You have to design and implement a frontend in React to support these operations.  
+
+You may work on the React project from before. You will have to implement few new pages:
+
+1. Product List page
+2. Product Page
+3. Create a new Product Page
+4. Update a Product Page
+
+Note that the API is not protected by any authentication mechanism so anyone can change or update the database. This is generally a bad idea but we'll keep it that way for simplicity. 
+
+You are free to design the pages as you wish but they should be consistent in color palette with the previous given design. You are free to choose any layout as you wish for each page. For inspiration checkout popular e-commerce websites like Amazon, Flipkart, etc. 
+
+## STEPS   
+
+
+1. Wireframe your designs in Figma. It is generally a good idea to do so. 
+2. Read the API specs carefully and figure out what fields you will get for each object. This is called the `schema` of the `model` in backend parlance.  
+
+The product model is given below along with their data types:
+
+```json
+ {
+    "title": "string",
+    "category": "string",
+    "price": 0,
+    "location": "string",
+    "description": "string",
```

Comment:
Added `img_url` and `metadata_user` fields
https://github.com/as1605/MarketPlace-Demo-Backend/commit/8d49742b91ac3402e9bafc0432cbcf43327ab9db

Reply:
accha ok will update this also